src/server/raftkv/mod.rs

```diff
@@ -842,7 +842,12 @@ impl ReadIndexObserver for ReplicaReadLockChecker {
             let begin_instant = Instant::now();
 
             let start_ts = request.get_start_ts().into();
-            self.concurrency_manager.update_max_ts(start_ts);
+            if let Err(e) = self
+                .concurrency_manager
+                .update_max_ts(start_ts, || format!("read_index-{}", start_ts))
+            {
+                error!("failed to update max ts in concurrency manager"; "err" => ?e);
```

Comment:
An error message needs to be returned to the follower peer, prompting it to reject this follower read as it might be unsafe.  Pritting error logs on the leader peer seems not to be enough.

Reply:
It requires modifying kvproto. This shall be done in another PR, together with introducing an error type for InvalidMaxTsUpdate.